modules/olm-restricted-networks-configuring-operatorhub.adoc

```diff
@@ -29,243 +37,40 @@ $ oc patch OperatorHub cluster --type json \
-+
-----
-$ curl https://quay.io/cnr/api/v1/packages?namespace=redhat-operators > packages.txt
-$ curl https://quay.io/cnr/api/v1/packages?namespace=community-operators >> packages.txt
-$ curl https://quay.io/cnr/api/v1/packages?namespace=certified-operators >> packages.txt
-----
-+
-Each package in the new `packages.txt` is an Operator that you could add to your
-restricted network catalog. From this list, you could either pull every Operator
-or a subset that you would like to expose to users.
-
-. **Pull Operator content.**
-+
-For a given Operator in the package list, you must pull its content:
-+
-----
-$ curl https://quay.io/cnr/api/v1/packages/<namespace>/<operator_name>/<release>
-----
-+
-This example uses the etcd Operator:
-
-.. Retrieve the digest:
-+
-----
-$ curl https://quay.io/cnr/api/v1/packages/community-operators/etcd/0.0.12
-----
-
-.. From that JSON, take the digest and use it to pull the gzipped archive:
-+
-----
-$ curl -XGET https://quay.io/cnr/api/v1/packages/community-operators/etcd/blobs/sha256/8108475ee5e83a0187d6d0a729451ef1ce6d34c44a868a200151c36f3232822b \
-    -o etcd.tar.gz
-----
-
-.. To pull the information out, you must untar the archive into a
-`manifests/<operator_name>/` directory with all the other Operators that you
-want. For example, to untar to an existing directory called `manifests/etcd/`:
+. Extract the contents of your custom Operator catalog image to generate manifests
+required for mirroring:
 +
 ----
-$ mkdir -p manifests/etcd/ <1>
-$ tar -xf etcd.tar.gz -C manifests/etcd/
-----
-<1> Create different subdirectories for each extracted archive so that files are not
-overwritten by subsequent extractions for other Operators.
+$ oc adm catalog mirror \
+    <registry_host_name>:<port>/<namespace>/redhat-operators:v1 \ <1>
+    <registry_host_name>:<port>/openshift <2>
 
-. *Break apart `bundle.yaml` content, if necessary.*
-+
-In your new `manifests/<operator_name>` directory, the goal is to get your bundle in the following directory structure:
-+
+mirroring ...
 ----
-manifests/
-└── etcd
-    ├── 0.0.12
-    │   ├── clusterserviceversion.yaml
-    │   └── customresourcedefinition.yaml
-    └── package.yaml
-----
-+
-If you see files already in this structure, you can skip this step. However, if
-you instead see only a single file called `bundle.yaml`, you must first break
-this file up to conform to the required structure.
-+
-You must separate the CSV content under `data.clusterServiceVersion` (each file
-in the list), the CRD content under `data.customResourceDefinition` (each file
-in the list), and the package content under `data.Package` into their own files.
-
-.. For the CSV file creation, find the following lines in the `bundle.yaml` file:
+<1> Specify your Operator catalog image.
+<2> Specify a namespace to mirror all referenced images.
 +
-[source,yaml]
-----
-data:
-  clusterServiceVersions: |
-----
+This generates an ImageContentSourcePolicy object that can configure nodes to
```

Comment:
@adellape See my comments https://bugzilla.redhat.com/show_bug.cgi?id=1773821#c21. I don't think it's available. Waiting for the developers response.

Reply:
Updated based on the outcome of the now-VERIFIED BZ.